src/Service/MappingService.php

```diff
@@ -227,21 +227,49 @@ private function handleCast(Dot $dotArray, string $key, string $cast)
 
         // Todo: Add more casts.
         switch ($cast) {
-        case 'int':
-        case 'integer':
-            $value = (int) $value;
+        case 'string':
+            $value = (string) $value;
             break;
         case 'bool':
         case 'boolean':
-            if ((int) $value === 1 || $value === 'true' || $value === 'True' || $value === 'TRUE') {
+            if ((int) $value === 1 || strtolower($value) === 'true' || strtolower($value) === 'yes') {
                 $value = true;
                 break;
             }
 
             $value = false;
             break;
-        case 'string':
-            $value = (string) $value;
+        case 'int':
+        case 'integer':
+            $value = (int) $value;
+            break;
+        case 'float':
+            $value = (float) $value;
+            break;
+        case 'array':
+            $value = (array) $value;
```

Comment:
Does this work?

Reply:
not sure, but it uses a cast described here, added this because it was present in de documentation but not in code. If someone wants to use this cast it is now available :) 
https://www.php.net/manual/en/language.types.type-juggling.php